### ISSUE TITLE
[Bug] [connector-redis] hash structure will report errors

### ISSUE THREAD

**Issue by lm-ylj** [apache/seatunnel#7774]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

I encountered an error when using MySQL as the source and Redis Hash type as the sink

### SeaTunnel Version

2.3.3

### SeaTunnel Config

```conf
env {
  execution.parallelism = 2
  job.mode = "BATCH"
  checkpoint.interval = 10000
}

source {
  Jdbc {
        url = "jdbc:mysql://xxx:3306/sea_tunnel_test?serverTimezone=GMT%2b8"
        driver = "com.mysql.cj.jdbc.Driver"
        connection_check_timeout_sec = 100
        user = "root"
        password = "xxx"
        query = "select * from t_user"
        partition_column = "id"
        split.size = 10000
  }
}

sink {
  Redis {
    host = "xxx"
    port = 6379
    key = "user_name"
    data_type = "hash"
    auth = "xxx"
  }
}
```

### Running Command

```shell
sh bin/seatunnel.sh --config config/mysql_to_redis_hash
```

### Error Exception

```log
StackTrace:org.apache.seatunnel.core.starter.exception.CommandExecuteException: SeaTunnel job executed failed
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:191)
        at org.apache.seatunnel.core.starter.SeaTunnel.run(SeaTunnel.java:40)
        at org.apache.seatunnel.core.starter.seatunnel.SeaTunnelClient.main(SeaTunnelClient.java:34)
Caused by: org.apache.seatunnel.engine.common.exception.SeaTunnelEngineException: java.lang.RuntimeException: redis.clients.jedis.exceptions.JedisDataException: ERR wrong number of arguments for 'hset' command
        at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:232)
        at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:61)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTransformCollector.collect(SeaTunnelTransformCollector.java:39)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTransformCollector.collect(SeaTunnelTransformCollector.java:27)
        at org.apache.seatunnel.engine.server.task.group.queue.IntermediateBlockingQueue.handleRecord(IntermediateBlockingQueue.java:76)
        at org.apache.seatunnel.engine.server.task.group.queue.IntermediateBlockingQueue.collect(IntermediateBlockingQueue.java:51)
        at org.apache.seatunnel.engine.server.task.flow.IntermediateQueueFlowLifeCycle.collect(IntermediateQueueFlowLifeCycle.java:52)
        at org.apache.seatunnel.engine.server.task.TransformSeaTunnelTask.collect(TransformSeaTunnelTask.java:73)
        at org.apache.seatunnel.engine.server.task.SeaTunnelTask.stateProcess(SeaTunnelTask.java:168)
        at org.apache.seatunnel.engine.server.task.TransformSeaTunnelTask.call(TransformSeaTunnelTask.java:78)
        at org.apache.seatunnel.engine.server.TaskExecutionService$BlockingWorker.run(TaskExecutionService.java:613)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: redis.clients.jedis.exceptions.JedisDataException: ERR wrong number of arguments for 'hset' command
        at redis.clients.jedis.Protocol.processError(Protocol.java:96)
        at redis.clients.jedis.Protocol.process(Protocol.java:137)
        at redis.clients.jedis.Protocol.read(Protocol.java:192)
        at redis.clients.jedis.Connection.readProtocolWithCheckingBroken(Connection.java:316)
        at redis.clients.jedis.Connection.getOne(Connection.java:298)
        at redis.clients.jedis.Connection.executeCommand(Connection.java:123)
        at redis.clients.jedis.Jedis.hset(Jedis.java:5548)
        at org.apache.seatunnel.connectors.seatunnel.redis.config.RedisDataType$2.set(RedisDataType.java:47)
        at org.apache.seatunnel.connectors.seatunnel.redis.sink.RedisSinkWriter.write(RedisSinkWriter.java:63)
        at org.apache.seatunnel.connectors.seatunnel.redis.sink.RedisSinkWriter.write(RedisSinkWriter.java:35)
        at org.apache.seatunnel.engine.server.task.flow.SinkFlowLifeCycle.received(SinkFlowLifeCycle.java:227)
        ... 15 more

        at org.apache.seatunnel.engine.client.job.ClientJobProxy.waitForJobComplete(ClientJobProxy.java:122)
        at org.apache.seatunnel.core.starter.seatunnel.command.ClientExecuteCommand.execute(ClientExecuteCommand.java:184)
        ... 2 more
```

### Zeta or Flink or Spark Version

Zeta

### Java or Scala Version

Java 1.8

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by FuYouJ** [comment]
@lm-ylj 
1：From Redis 4.0 onwards,HMSET is considered obsolete and HSET is recommended instead。
see https://redis.io/docs/latest/commands/hmset/
2：I tried to reproduce the issue you mentioned, and I believe the problem does not lie in the fact that HSET cannot set values in bulk. After testing, I found that HSET can indeed set values in batch.
![QQ_1727945522662](https://github.com/user-attachments/assets/b1468f3b-40d4-4a2a-bca6-0cf007e24a9a)
3 : The error mentioned in the issue occurs when the map is empty.
![QQ_1727945692116](https://github.com/user-attachments/assets/d143ad03-d17b-4e2d-9d1d-32564d363a91)
4 : So I believe the issue might lie in other aspects. More input and output references are needed to determine the problem. such as input data,redis version
